### ISSUE TITLE
AVD 3.7.0 using network_ports on spine requires to activate some network-services

### ISSUE THREAD

**Issue by gmuloc** [aristanetworks/ansible-avd#2234]
### Issue Summary

Currently it is only possible to leverage `network_ports` abstraction on `node_type_keys` where at least one network-service is enabled.

The use case is to be able to shutdown and add a description on unused ports but when enabling connected_endpoints the logic is not run.

e.g.
```
---
# node types for the deployment
node_type_keys:
  spine:
    type: spine
    default_evpn_role: server
    # Overwriting default interface_descriptions template for underlay to allow
    # lower case
    interface_descriptions:
      underlay_ethernet_interfaces: 'custom_templates/interface_descriptions/underlay/ethernet_interfaces.j2'
    # Enabling connected_endpoints for Spines to be able to leverage
    # network_ports abstraction for unused ports
    connected_endpoints: true
    # Temporary workaround where network_ports can only be used when at least
    # one network service is enabled
    network_services:
      l1: true
```

then in the spine group_vars file:

```
# The breakout ports are marked as breakout unusable
network_ports:
  - switches:
      - [...]
    switch_ports:
      - [...]
    profile: unused_port
[...]
# Necessary to prevent patch_panel config that is auto-generated
# based on the workaround to enable network_ports for Spines
custom_group_config_patch_panel: null
```

### Which component(s) of AVD impacted

eos_designs

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

cf description

The issue is that without any network-service enabled the following error is yielded by AVD (output with `-vvvvv`):

```
fatal: [device1 -> localhost]: FAILED! => {
    "msg": "'dict object' has no attribute 'enable_trunk_groups'"
}
fatal: [device2 -> localhost]: FAILED! => {
    "msg": "'dict object' has no attribute 'enable_trunk_groups'"
}
```

The reason is quite convoluted.

If no service is enable then https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py#L546 is False

and then this means that enable_trunk_groups is None: https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py#L594 
and removed from the facts at the end.

This leads to the error message above when the key is tried to be accessed in the connected-endpoints/ethernet-interfaces.j2 template:
https://github.com/aristanetworks/ansible-avd/blob/v3.7.0/ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2#L91

Throwing the error because the template is not checking if the variable is enabled.

### Relevant log output

cf above.

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

**Comment by gmuloc** [comment]
Note the same content runs fine in 3.8.0-dev2

**Comment by gmuloc** [comment]
This has been fixed in #2038
